### ISSUE TITLE
quiet EzCards should be a11y friendly

### ISSUE THREAD

**Issue by djcp** [ezcater/recipe#692]
## Describe the bug

Peter Grippi said in a slack DM:

> I also noticed in my testing that \<EzCard quiet clickable /\> is not a11y friendly, we can’t select cards using tab and use the keyboard to open them, so there’s probably some more accessibility work needed in recipe if we end up using it for the menu item cards

## Steps To Reproduce

1.
2.

## Current behavior

ezCards that are quiet and clickable can't be selected via tabbing, making it impossible to click them.

## Expected behavior

ezCards that are quiet and clickable *can* be selected via tabbing, making them clickable and a11y friendly.

## Additional context

I'm reporting this by proxy for @pgrippi

**Comment by CraigCav** [comment]
@pgrippi I'm able to use my keyboard to navigate to and interact with clickable cards on the [doc-site example](https://ezcater.github.io/recipe/components/ez-card#clickable-cards).

![clickable-cards demo showing keyboard support for selecting and interacting with cards](https://user-images.githubusercontent.com/109814/129051324-556e745e-02bc-4a09-a9b3-4b360c7754ad.gif)

Could you elaborate on what is missing in terms of being a11y friendly?

**Comment by pgrippi** [comment]
Can you look at this file and tell me what I'm doing wrong? https://github.com/ezcater/store-next/blob/bf409a6fc9c06a9c7cbed4797ae45dd89765dc09/src/pageComponents/catering-menu/RecipeMenuItemCard/RecipeMenuItemCard.tsx#L34

When I tab it ignores all of the cards and only focuses on the buttons on the page:


https://user-images.githubusercontent.com/1180841/129053013-2e9e8940-6f38-42e6-90c2-464f53969ab3.mp4

**Comment by pgrippi** [comment]
After looking at the example I have a feeling the "what I'm doing wrong" is that there is no link inside the card and so it's not being considered as a focusable container. If that's the case is there a suggestion for what we should do when there isn't a link? (for this use case it doesn't make sense to have a link)

**Comment by CraigCav** [comment]
@pgrippi You're spot on; there's no interactive element within the card to focus on, so focus is skipping to the next interactive element.

Recipe follows [this guidance on creating accessible cards](https://inclusive-components.design/cards/). The premise really is that the card is a "large click surface" for some underlying action, like navigation or clicking a button.

I'd make the argument that menu item cards are conceptually links to view a menu item, albeit we enhance the experience client side by popping up the item in-context in a dialog.  In which case, I'd suggest that the Item name is a link (it's even styled like a link in the menu itself!).

That being said, you could also use a call-to-action button (similar to [this example](https://ezcater.github.io/recipe/components/ez-card#card-with-a-specific-call-to-action) - based on [this guidance](https://inclusive-components.design/cards/#callstoaction)), which can offer some more explicit instructions to indicate that the card is interactive.

One last alternative that works, is making the card focusable, using `tabIndex`. I'd caution that if you go this route, you should also provide a `role` attribute, since the card is effectively acting like a button. You might also want to consider that screen readers will then read out the entire content of the card on focus, which isn't necessarily what you might want.

**Comment by pgrippi** [comment]
> One last alternative that works, is making the card focusable, using tabIndex. I'd caution that if you go this route, you should also provide a role attribute, since the card is effectively acting like a button. You might also want to consider that screen readers will then read out the entire content of the card on focus, which isn't necessarily what you might want.

Yeah we went down this path in store-react, and it is a PITA. We also need to set key listeners to check for specific events. It's a lot easier to just use a button or a link which comes with all of that built in natively

> I'd make the argument that menu item cards are conceptually links to view a menu item, albeit we enhance the experience client side by popping up the item in-context in a dialog. In which case, I'd suggest that the Item name is a link (it's even styled like a link in the menu itself!).

The qualm I have with this is that the link doesn't actually go anywhere, there is no page that displays the menu item. So if we went this route are we back to the jquery days of doing `<a href="#">` and preventing the default action so the browser doesn't navigate? If so I think I'd rather go the button route

Either way I think this issue can be closed since this is user error!
